### PR TITLE
Updated CarrierAccount delete by ID

### PR DESCRIPTION
**Pull request by algreen11**
No longer have to retrieve CarrierAccount before you can delete #15 

pom changed to 4.11 #30

### CODE REVIEW

**Comment by algreen11** [comment]
The java documentation on easypost needs to change. 

It is currently :
import com.easypost.EasyPost;
import com.easypost.exception.EasyPostException;

EasyPost.apiKey = "<YOUR_PRODUCTION_API_KEY>";

CarrierAccount ca = CarrierAccount.retrieve("ca_...");
ca.delete();

but can now be updated to :
import com.easypost.EasyPost;
import com.easypost.exception.EasyPostException;

EasyPost.apiKey = "<YOUR_PRODUCTION_API_KEY>";

CarrierAccount.delete("ca_...");

File: pom.xml [hunk 1]
```diff
@@ -49,7 +49,7 @@
         <dependency>
             <groupId>junit</groupId>
             <artifactId>junit</artifactId>
-            <version>4.10</version>
+            <version>4.11</version>
```
Comment (victoryftw) [inline pom.xml]: Are we intentionally trying to increase the version of Junit? That sounds like a different kind of change than the rest of whats done here. I think you want to be the version on line 8 instead.

File: src/main/java/com/easypost/model/CarrierAccount.java [hunk 2]
```diff
@@ -57,11 +57,16 @@ public CarrierAccount update(Map<String, Object> params, String apiKey) throws E
...
+			if(id.substring(0,2).equals("ca")){
+				return delete(id,null); 
+			}
+			else{
+				return null;
+			}
 		}
-		public void delete(String apiKey) throws EasyPostException {
-			request(RequestMethod.DELETE, instanceURL(CarrierAccount.class, this.getId()), null, CarrierAccount.class, apiKey);
+		public static CarrierAccount delete(String id, String apiKey) throws EasyPostException {
```
Comment (victoryftw) [inline src/main/java/com/easypost/model/CarrierAccount.java]: I believe we still want this to be a `void` method, not returning a `CarrierAccount`. Once the carrier account is deleted, what is there to return?

File: src/main/java/com/easypost/model/CarrierAccount.java [hunk 3]
```diff
@@ -57,11 +57,16 @@ public CarrierAccount update(Map<String, Object> params, String apiKey) throws E
...
+			return request(RequestMethod.DELETE, instanceURL(CarrierAccount.class, id), null, CarrierAccount.class, apiKey);
```
Comment (victoryftw) [inline src/main/java/com/easypost/model/CarrierAccount.java]: The CarrierAccount::Delete endpoint in EasyPost returns empty JSON, so I don't actually think there's anything here to return.

File: pom.xml (hunk 1 above)
Comment (algreen11) [inline pom.xml]: This fixes a compilation error (#30) when initially building the library (at least it did for me).

**Comment by algreen11** [comment]
Thats a good idea, I think we should keep both. Will update it now.

**Pull request closed without merging**